make check failing test cases. (Linux 64-Bit Ubuntu 16.04)

I am building from the master branch of universal-ctags.
Below are the configuration, build and check commands, I currently use(since I don't have root access on the system.):

```sh
# From universal-ctags source directory
$ mkdir objdir
$ cd objdir
$ ../configure --prefix=$HOME/apps/ctags --program-prefix=u_
$ make -j4
$ make check
``` 

Below is the truncated log of the make check output:

```text
Summary (see CMDLINE.tmp to reproduce without test harness)
------------------------------------------------------------
  #passed:                                720
  #FIXED:                                 0
  #FAILED (unexpected-exit-status):       0
  #FAILED (unexpected-output):            21
	parser-c.r/bit_field.c
	parser-c.r/bug1466117.c
	parser-c.r/bug1491666.c
	parser-c.r/bug556646.c
	parser-c.r/c-struct-var-with-initializer
	parser-c.r/enum.c
	parser-cxx.r/bug639639.cpp
	parser-cxx.r/bug639644.cpp
	parser-cxx.r/cxx11-lambdas
	parser-cxx.r/cxx11enum.cpp
	parser-cxx.r/namespace.cpp
	parser-cxx.r/templates-in-labmdas-1.cpp
	parser-cxx.r/templates-in-labmdas-2.cpp
	parser-cxx.r/typedefs.cpp
	parser-javascript.r/js-complex-return
	parser-javascript.r/js-es6-class
	parser-javascript.r/js-es6-mixin
	parser-javascript.r/js-many-functions
	parser-yacc.r/c-anon-ids
	parser-yacc.r/nested
	xformat-option.r/format-NlKkFnP
  #TIMED-OUT (10s)                        0
  #skipped (features):                    1
	parser-ant.r/regex-based
  #skipped (languages):                   4
	main-xcmd.r/xcmd-coffeetags-pseudo
	main-xcmd.r/xcmd-coffeetags
	main-xcmd.r/xcmd-etags
	main-xcmd.r/xcmd-xref
  #known-bugs:                            12
	parser-autoconf.r/nested-block.ac
	parser-c.r/attr-attached-to-array-failed
	parser-c.r/extern_variable.h
	parser-cxx.r/cxx-keywords-as-c-identifiers
	parser-cxx.r/cxx-keywords-as-c-identifiers
	parser-cxx.r/namespace-and-scope
	parser-cxx.r/variables-prototypes.cpp
	parser-java.r/bug1777344.java
	parser-m4.r/m4-quotestring
	parser-python.r/newlines-cr
	parser-scheme.r/scheme-srfi-30-comment
	parser-scheme.r/scheme-string
Makefile:6271: recipe for target 'units' failed
make: *** [units] Error 1
```

Sorry for populating the issue page, since I could not find the discussion portal (gitter/irc etc.).

**Note:** I have installed the following dependencies:
```
$ sudo apt install libyaml-dev libxml2-dev libjansson-dev
```

Let me know if I have missed any other dependencies.

-Ameya

**Note:** Updated the build steps for object directory for out of source tree build.

--- a/main/options.c
+++ b/main/options.c
@@ -421,6 +421,8 @@ static optionDescription LongOptionDescription [] = {
  {1,"       Allow specifying --xcmd-<LANG> option in ~/.ctags and/or ~/.ctags/*."},
  {1,"       By default it is not allowed. This option itself can be specified only"},
  {1,"       in /etc or /usr/local/etc."},
+ {1,"  --_anonhash=fname"},
+ {1,"       Used in u-ctags test harness"},
  {1,"  --_echo=msg"},
  {1,"       Echo MSG to standard error. Useful to debug the chain"},
  {1,"       of loading option files."},
@@ -2339,6 +2341,17 @@ static void processIgnoreOption (const char *const list, int IgnoreOrDefine)
 		readIgnoreList (list);
 }
 
+static void processAnonHashOption (const char *const option, const char *const parameter)
+{
+	if (parameter == NULL || parameter[0] == '\0')
+		error (FATAL, "Something string is needed for \"%s\" option", option);
+	char buf [9];
+
+	anonHashString (parameter, buf);
+	printf("%s\n", buf);
+	exit (0);
+}
+
 static void processEchoOption (const char *const option, const char *const parameter)
 {
 	if (parameter == NULL || parameter[0] == '\0')
@@ -2573,6 +2586,7 @@ static parametricOption ParametricOptions [] = {
 	{ "sort",                   processSortOption,              true,   STAGE_ANY },
 	{ "tag-relative",           processTagRelative,             true,   STAGE_ANY },
 	{ "version",                processVersionOption,           true,   STAGE_ANY },
+	{ "_anonhash",              processAnonHashOption,          false,  STAGE_ANY },
 	{ "_echo",                  processEchoOption,              false,  STAGE_ANY },
 	{ "_force-quit",            processForceQuitOption,         false,  STAGE_ANY },
 	{ "_xformat",               processXformatOption,           false,  STAGE_ANY },

--- a/main/parse.c
+++ b/main/parse.c
@@ -3221,17 +3221,24 @@ static unsigned int anonHash(const unsigned char *str)
 	return hash ;
 }
 
+extern void anonHashString (const char *filename, char buf[9])
+{
+	sprintf(buf, "%08x", anonHash((const unsigned char *)filename));
+}
+
+
 extern void anonGenerate (vString *buffer, const char *prefix, int kind)
 {
 	parserObject* parser = LanguageTable + getInputLanguage ();
 	parser -> anonymousIdentiferId ++;
 
 	char szNum[32];
+	char buf [9];
 
 	vStringCopyS(buffer, prefix);
 
-	unsigned int uHash = anonHash((const unsigned char *)getInputFileName());
-	sprintf(szNum,"%08x%02x%02x",uHash,parser -> anonymousIdentiferId, kind);
+	anonHashString (getInputFileName(), buf);
+	sprintf(szNum,"%s%02x%02x",buf,parser -> anonymousIdentiferId, kind);
 	vStringCatS(buffer,szNum);
 }
 

--- a/main/parse.h
+++ b/main/parse.h
@@ -242,5 +242,6 @@ extern bool makeKindDescriptionsPseudoTags (const langType language,
 					       const ptagDesc *pdesc);
 
 extern void anonGenerate (vString *buffer, const char *prefix, int kind);
+extern void anonHashString (const char *filename, char buf[9]);
 
 #endif  /* CTAGS_MAIN_PARSE_H */

--- a/misc/units
+++ b/misc/units
@@ -420,6 +420,20 @@ direq ()
     return $?
 }
 
+anon_normalize ()
+{
+    local ctags=$1
+    local input_actual=$2
+
+    # TODO: "Units" should not be hardcoded.
+    local input_expected="./Units${input_actual#*/Units}"
+
+    local actual=$(${CTAGS} --quiet --options=NONE --_anonhash="${input_actual}")
+    local expected=$(${CTAGS} --quiet --options=NONE --_anonhash="${input_expected}")
+
+    sed -e s/${actual}/${expected}/g
+}
+
 run_tcase ()
 {
     local input="$1"
@@ -627,7 +641,9 @@ run_tcase ()
 	return 0
     fi
 
-    ${output_type}_basename_filter < "${orawout}" | "$ffilter" > "${ofiltered}"
+    ${output_type}_basename_filter < "${orawout}" | \
+	anon_normalize "${CTAGS}" "${input}" | \
+	$ffilter > "${ofiltered}"
 
     {
 	diff -U 0 -I '^!_TAG' --strip-trailing-cr "${fexpected}" "${ofiltered}" > "${odiff}"

